core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala

```diff
@@ -92,26 +84,67 @@ abstract class IntegrationTestHarness extends KafkaServerTestHarness {
       clientSaslProperties)
   }
 
-  def createProducer: KafkaProducer[Array[Byte], Array[Byte]] = {
-    TestUtils.createProducer(brokerList,
-      securityProtocol = this.securityProtocol,
-      trustStoreFile = this.trustStoreFile,
-      saslProperties = this.clientSaslProperties,
-      props = Some(producerConfig))
+  def createProducer[K, V](acks: Int = -1,
+                           maxBlockMs: Long = 60 * 1000L,
+                           bufferSize: Long = 1024L * 1024L,
+                           retries: Int = Int.MaxValue,
+                           deliveryTimeoutMs: Int = 30 * 1000,
+                           lingerMs: Int = 0,
+                           requestTimeoutMs: Int = 20 * 1000,
+                           keySerializer: Serializer[K] = new ByteArraySerializer,
+                           valueSerializer: Serializer[V] = new ByteArraySerializer,
+                           configOverrides: Properties = new Properties()): KafkaProducer[K, V] = {
```

Comment:
This is kind of annoying. We should probably decouple getting the configs from creating the producer/consumer. Then we can avoid this issue, I think?

Reply:
Ack. I think we should just avoid the optional arguments. They are rarely used anyway and they make this api confusing.